e2e/azure_test.go

```diff
@@ -0,0 +1,268 @@
+	By("Creating Azure cluster secret")
+	capzManagerBootstrapCredentialsKey := client.ObjectKey{Namespace: framework.CAPINamespace, Name: capzManagerBootstrapCredentials}
+	capzManagerBootstrapCredentials := &corev1.Secret{}
+
+	if err := cl.Get(ctx, capzManagerBootstrapCredentialsKey, capzManagerBootstrapCredentials); err != nil {
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	azureClientSecret, found := capzManagerBootstrapCredentials.Data["azure_client_secret"]
+	Expect(found).To(BeTrue())
+
+	azureSecretKey := corev1.SecretReference{Name: clusterSecretName, Namespace: framework.CAPINamespace}
+	azureSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      azureSecretKey.Name,
+			Namespace: azureSecretKey.Namespace,
+		},
+		Immutable: ptr.To(true),
+		Data: map[string][]byte{
+			"clientSecret": azureClientSecret,
+		},
+	}
+
+	if err := cl.Create(ctx, &azureSecret); err != nil && !apierrors.IsAlreadyExists(err) {
+		Expect(err).ToNot(HaveOccurred())
+	}
+	By("Creating Azure cluster identity")
+
+	var azureClientID, azureTenantID []byte
+	azureClientID, found = capzManagerBootstrapCredentials.Data["azure_client_id"]
+	Expect(found).To(BeTrue())
+	azureTenantID, found = capzManagerBootstrapCredentials.Data["azure_tenant_id"]
+	Expect(found).To(BeTrue())
+
+	azureClusterIdentity := &azurev1.AzureClusterIdentity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: framework.CAPINamespace,
+			Annotations: map[string]string{
+				// The ManagedBy Annotation is set so CAPI infra providers ignore the InfraCluster object,
+				// as that's managed externally, in this case by the cluster-capi-operator's infracluster controller.
+				clusterv1.ManagedByAnnotation: managedByAnnotationValueClusterCAPIOperatorInfraClusterController,
+			},
+		},
+		Spec: azurev1.AzureClusterIdentitySpec{
+			Type:              azurev1.ServicePrincipal,
+			AllowedNamespaces: &azurev1.AllowedNamespaces{NamespaceList: []string{framework.CAPINamespace}},
+			ClientID:          string(azureClientID),
+			TenantID:          string(azureTenantID),
+			ClientSecret:      corev1.SecretReference{Name: clusterSecretName, Namespace: framework.CAPINamespace},
+		},
+	}
+
+	if err := cl.Create(ctx, azureClusterIdentity); err != nil && !apierrors.IsAlreadyExists(err) {
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	By("Creating Azure cluster")
+	azureCluster := &azurev1.AzureCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: framework.CAPINamespace,
+		},
+		Spec: azurev1.AzureClusterSpec{
+			AzureClusterClassSpec: azurev1.AzureClusterClassSpec{
+				Location:         mapiProviderSpec.Location,
+				AzureEnvironment: "AzurePublicCloud",
+				IdentityRef: &corev1.ObjectReference{
+					Name:      clusterName,
+					Namespace: framework.CAPINamespace,
+					Kind:      "AzureClusterIdentity",
+				},
+			},
```

Comment:
Is this required for things to work? I was thinking ClusterClass is not a requirement